src/cycles/basis.jl

```diff
@@ -0,0 +1,72 @@
+# Code in this file inspired by NetworkX.
+
+"""
+    cycle_basis(g, root=nothing)
+
+Returns a list of cycles which form a basis for cycles of `g`.
```

Comment:
For docstrings, we use action words: "Return a list of cycles....". Also, explain the arguments here:

> Return a list of cycles which form a basis for cycles of graph `g`, optionally starting at node `root`.